sa/sa.go

```diff
@@ -213,49 +214,7 @@ func (ssa *SQLStorageAuthority) GetAuthorization(ctx context.Context, id string)
 // GetValidAuthorizations returns the latest authorization object for all
 // domain names from the parameters that the account has authorizations for.
 func (ssa *SQLStorageAuthority) GetValidAuthorizations(ctx context.Context, registrationID int64, names []string, now time.Time) (map[string]*core.Authorization, error) {
-	if len(names) == 0 {
-		return nil, berrors.InternalServerError("no names received")
-	}
-
-	params := make([]interface{}, len(names))
-	qmarks := make([]string, len(names))
-	for i, name := range names {
-		id := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: name}
-		idJSON, err := json.Marshal(id)
-		if err != nil {
-			return nil, err
-		}
-		params[i] = string(idJSON)
-		qmarks[i] = "?"
-	}
-
-	auths, err := selectAuthzs(ssa.dbMap,
-		"WHERE registrationID = ? "+
-			"AND expires > ? "+
-			"AND identifier IN ("+strings.Join(qmarks, ",")+") "+
-			"AND status = 'valid'",
-		append([]interface{}{registrationID, now}, params...)...)
-	if err != nil {
-		return nil, err
-	}
-
-	byName := make(map[string]*core.Authorization)
-	for _, auth := range auths {
-		// No real life authorizations should have a nil expires. If we find them,
-		// don't consider them valid.
-		if auth.Expires == nil {
-			continue
-		}
-		if auth.Identifier.Type != core.IdentifierDNS {
-			return nil, fmt.Errorf("unknown identifier type: %q on authz id %q", auth.Identifier.Type, auth.ID)
-		}
-		existing, present := byName[auth.Identifier.Value]
-		if !present || auth.Expires.After(*existing.Expires) {
-			byName[auth.Identifier.Value] = auth
-		}
-	}
-
-	return byName, nil
+	return ssa.getAuthorizations(ctx, "authz", string(core.StatusValid), registrationID, names, now)
```

Comment:
Is there a constant we can use for the table names to not duplicate the "authz"/"pendingAuthz" values for callers of `getAuthorizations`?